server/datasources/memory.datasource.js

```diff
@@ -1,21 +1,4 @@
-"use strict";
+'use strict';
 
-const { decorate, metadata, param } = require('../../lib/helpers.js');
-const core = require("@loopback/core");
-const repository = require("@loopback/repository");
-const config = require("./memory.datasource.json");
-
-class MemoryDataSource extends repository.juggler.DataSource {
-  constructor(dsConfig = config) {
-    super(dsConfig);
-  }
-};
-
-MemoryDataSource.dataSourceName = 'memory';
-
-MemoryDataSource = decorate([
-  param(0, core.inject('datasources.config.memory', { optional: true })),
-  metadata("design:paramtypes", [Object])
-], MemoryDataSource);
-
-exports.MemoryDataSource = MemoryDataSource;
+const config = require('./memory.datasource.json');
+exports.MemoryDataSource = datasourceFactory('memory', config);
```

Comment:
> I'd much rather find a clean solution that works both for TS and JS codebases, rather than trying to implement a datasource-specific hack like what is proposed here.

Or even better, find out an easy-to-use solution for configuring Dependency Injection from JavaScript (because we need it anyway) and then we can keep the existing code structure.